src/withStyles.jsx

```diff
@@ -51,28 +51,47 @@ export function withStyles(
   /** Cache for storing the result of stylesFn(theme) for all themes. */
   const stylesFnResultCacheMap = typeof WeakMap === 'undefined' ? new Map() : new WeakMap();
```

Comment:
The WeakMap would only have an impact if the theme was garbage collectable, but since the withStylesCache is not a weak map it'll never be collectable. We should use a WeakMap for both if we want dynamically allocated themes to not leak.